lesson7/index.js

```diff
@@ -0,0 +1,39 @@
+const ul = document.getElementById('js-ul');
+
+const data = [
+  { to: "bookmark.html", img: "1.png", alt: "画像1", text: "ブックマーク" },
+  { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" }
+];
+
+function loading() {
+  const LoadingImage = document.createElement('img');
+  LoadingImage.src = "./loading-circle.gif";
+  ul.appendChild(LoadingImage);
+}
+
+function getData() {
+  loading();
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(data);
+    }, 1000);
+  });
+}
+
+getData().then((lists) => {
+  ul.removeChild(ul.firstChild);
+  const frag = document.createDocumentFragment();
+  lists.forEach((list) => {
+    const li = document.createElement('li');
+    const anchor = document.createElement('a');
+    const img = document.createElement('img');
+
+    anchor.href = `/${list.to}`;
+    img.src = list.img;
+    img.alt = list.alt;
+
+    frag.appendChild(li).appendChild(anchor).appendChild(img);
+    anchor.insertAdjacentHTML('beforeend', list.text);
+  });
+  ul.appendChild(frag);
+});
```

Comment:
Let's cut out the processes that can be cut out into functions and execute them.I learned from Mr. Morita that this is because it is easier to test later and the code can withstand specification changes.

There are two types of processing in .then, right?
IMO
```suggestion
getData().then((lists) => {
hogehoge();//Erasing an image
hugahuga();//Process of creating an element
});
```

Reply:
Thank you for the review.


> Let's cut out the processes that can be cut out into functions and execute them.I learned from Mr. Morita that this is because it is easier to test later and the code can withstand specification changes.

I learned what to pay attention to when writing code. Thank you very much.
e27ee45217d76eb4a062a50f5a7575471cda1f3e